src/Sema.zig

```diff
@@ -2636,10 +2634,12 @@ fn zirMergeErrorSets(sema: *Sema, block: *Scope.Block, inst: Zir.Inst.Index) Com
         };
         return sema.mod.failWithOwnedErrorMsg(&block.base, msg);
     }
-    if (rhs_ty.zigTypeTag() != .ErrorSet)
-        return sema.mod.fail(&block.base, rhs_src, "expected error set type, found {}", .{rhs_ty});
+    const lhs_ty = try sema.analyzeAsType(block, lhs_src, lhs);
+    const rhs_ty = try sema.analyzeAsType(block, rhs_src, rhs);
     if (lhs_ty.zigTypeTag() != .ErrorSet)
         return sema.mod.fail(&block.base, lhs_src, "expected error set type, found {}", .{lhs_ty});
+    if (rhs_ty.zigTypeTag() != .ErrorSet)
```

Comment:
Wont that eat a diagnostic if both are erroneous?

Reply:
should be fine. if you provide a test case, I'm happy to look at this idea in more detail. Either way, for the purposes of this branch, all we need to do is match master branch.